### PR TITLE
add a multi-task example

### PR DESCRIPTION
**Pull request by lyttonhao**
Ref to issue [#647](https://github.com/dmlc/mxnet/issues/647).

### CODE REVIEW

File: example/multi-task/example_multi_task.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    val_iter.reset()
+    start = time.time()
+    # train
+    for dbatch in train_iter:
+        data = dbatch.data[0]
+        label = dbatch.label[0]
+        # copy data and two labels into args
+        args["data"][:] = data 
+        args["softmax1_label"][:] = label
+        args["softmax2_label"][:] = label
```
Comment (piiswrong) [inline example/multi-task/example_multi_task.py]: Add comment here to explain that different labels should be used here for actual application.

File: example/multi-task/example_multi_task.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        executor.forward(is_train=True)
```
Comment (piiswrong) [inline example/multi-task/example_multi_task.py]: Could you see if DataParallelExecutorManager can be used here? It allows easier data loading and distributed training.

**Comment by piiswrong** [comment]
I made a few comments. Overall this looks good. Thanks for your contribution.

**Comment by piiswrong** [comment]
Actually, I think you can use FeedForward model for this. There is no need to write a custom solver.
That being said, a guide on how to write custom solvers is also welcomed. Could you add a switch to allow using either FeedForward or custom solver?

**Comment by lyttonhao** [comment]
@piiswrong Thanks a lot for your suggestions. But I'm not very sure about how to directly use DataParallelExecutorManager or FeedForward model for multi-task. It seems that I need to modify the IO iterator to specify the label names if using FeedForward model. Do I understand right? Thanks!

**Comment by tqchen** [comment]
Yes, I think we only need some kind of remapped data iter for multi label. It would be much better to use Feedforward model as we can easily take advantages of other things, e.g. multiGPU support

**Comment by tqchen** [comment]
@lyttonhao can you follow up on this? Basically it would be great if you can change this example to using FeedForward API

**Comment by lyttonhao** [comment]
@tqchen Ok. I will try to implement it by using FeedForward API.

**Comment by tqchen** [comment]
@lyttonhao Do we have any updates on this? Thanks!

**Comment by lyttonhao** [comment]
@tqchen Sorry about the delay. I will update it soon after a conference deadline.

**Comment by lyttonhao** [comment]
Replaced by #1659.

**Pull request closed without merging**